Testnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -282,7 +294,9 @@ public Address OwnerOf(UInt256 tokenId)
     /// <returns>Address that tokenId is approved for. </returns>
     public Address GetApproved(UInt256 tokenId)
     {
-        ValidateToken(tokenId);
+        Address tokenOwner = GetIdToOwner(tokenId);
+
+        EnsureAddressIsNotEmpty(tokenOwner);
```

Comment:
Is getting tokenOwner info necessary at all here?

Reply:
Yes because if you remove EnsureAddressIsNotEmpty(tokenOwner); then method will return false 

But method will throws exception in the case of token is not exist. 
So method behavior is changing when the assertion removed.

Reply:
Shouldn't an approval never be able to be set without an owner? So passing an Id to `GetIdToApproval` should return an address or `Address.Zero` if no approval, so maybe _that_ address should be checked for Empty, not the tokenOwner right? 

Saves some gas not getting the owner when we don't have too if we can just validate the Approval address prior to returning.

Reply:
These recommendation change some behavior of method. Both [OpenZeppelin](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/e63b09c9ad3a45484b6dc304e0e99640a9dc3036/contracts/token/ERC721/ERC721.sol#L127) and [0xcert](https://github.com/0xcert/ethereum-erc721/blob/77030ec2a25123d17b13e7ee469f1c956a48451c/src/contracts/tokens/nf-token.sol#L267) implementation is designed this way so i will stick to those implementations.